model.py

```diff
@@ -0,0 +1,176 @@
+
+DEBUG = False
+
+class SessionManager(object):
+
+    engine_for_url = {}
+
+    @classmethod
+    def engine(cls, url=None):
+        url = url or Configuration.database_url()
+        return create_engine(url, echo=DEBUG)
+
+    @classmethod
+    def sessionmaker(cls, url=None):
+        engine = cls.engine(url)
+        return sessionmaker(bind=engine)
+
+    @classmethod
+    def initialize(cls, url):
+        if url in cls.engine_for_url:
+            engine = cls.engine_for_url[url]
+            return engine, engine.connect()
+
+        engine = cls.engine(url)
+
+        Base.metadata.create_all(engine)
+
+        
+        cls.engine_for_url[url] = engine
+        return engine, engine.connect()
+
+    @classmethod
+    def session(cls, url):
+        engine = connection = 0
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=sa_exc.SAWarning)
+            engine, connection = cls.initialize(url)
+        session = Session(connection)
+        cls.initialize_data(session)
+        session.commit()
+        return session
+
+    @classmethod
+    def initialize_data(cls, session):
+        pass
+
+def get_one(db, model, on_multiple='error', **kwargs):
+    q = db.query(model).filter_by(**kwargs)
+    try:
+        return q.one()
+    except MultipleResultsFound, e:
+        if on_multiple == 'error':
+            raise e
+        elif on_multiple == 'interchangeable':
+            # These records are interchangeable so we can use
+            # whichever one we want.
+            #
+            # This may be a sign of a problem somewhere else. A
+            # database-level constraint might be useful.
+            q = q.limit(1)
+            return q.one()
+    except NoResultFound:
+        return None
+
+def get_one_or_create(db, model, create_method='',
+                      create_method_kwargs=None,
+                      **kwargs):
+    one = get_one(db, model, **kwargs)
+    if one:
+        return one, False
+    else:
+        __transaction = db.begin_nested()
+        try:
+            if 'on_multiple' in kwargs:
+                # This kwarg is supported by get_one() but not by create().
+                del kwargs['on_multiple']
+            obj = create(db, model, create_method, create_method_kwargs, **kwargs)
+            __transaction.commit()
+            return obj
+        except IntegrityError, e:
+            logging.info(
+                "INTEGRITY ERROR on %r %r, %r: %r", model, create_method_kwargs, 
+                kwargs, e)
+            __transaction.rollback()
+            return db.query(model).filter_by(**kwargs).one(), False
+
+def create(db, model, create_method='',
+           create_method_kwargs=None,
+           **kwargs):
+    kwargs.update(create_method_kwargs or {})
+    created = getattr(model, create_method, model)(**kwargs)
+    db.add(created)
+    db.flush()
+    return created, True
+
+    
+Base = declarative_base()
+
+class Place(Base):
```

Comment:
Except for this class and its test, all the code in this branch is a slightly modifed version of the same code in `core`.